### PR TITLE
platform-checks: Add migration scenarios, ci: Switch remaining jobs to ARM

### PR DESCRIPTION
**Pull request by def-**
I'll wait for nightly to finish with the changed jobs: https://buildkite.com/materialize/nightlies/builds/6204

### Checklist

- [ ] This PR has adequate test coverage / QA involvement has been duly considered.
- [ ] This PR has an associated up-to-date [design doc](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/README.md), is a design doc ([template](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/00000000_template.md)), or is sufficiently small to not require a design.
  
- [ ] If this PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way), then it is tagged with a `T-proto` label.
- [ ] If this PR will require changes to cloud orchestration or tests, there is a companion cloud PR to account for those changes that is tagged with the release-blocker label ([example](https://github.com/MaterializeInc/cloud/pull/5021)).
  
- [ ] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):
  -

### CODE REVIEW

File: ci/plugins/mzcompose/hooks/command [hunk 1]
```diff
@@ -80,4 +80,7 @@ fi
 
 ci_uncollapsed_heading ":docker: Running \`bin/mzcompose --find \"$BUILDKITE_PLUGIN_MZCOMPOSE_COMPOSITION\" run ${run_args[*]}\`"
 
+docker run --rm -t amd64/ubuntu uname -m
```
Comment (philip-stoev) [inline ci/plugins/mzcompose/hooks/command:83]: A comment here would be appreciated . those command lines have one too many unusual arguments

File: misc/python/materialize/checks/scenarios_migration.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+class MigrateX86Aarch64(Scenario):
+    """Migrate from x86-64 to aarch64 before the validation, Linux specific since we don't have x86-64 macOS images on Dockerhub"""
+
+    def actions(self) -> list[Action]:
+        return [
+            StartMz(self, platform="linux/amd64"),
+            Initialize(self),
+            Manipulate(self, phase=1),
+            Manipulate(self, phase=2),
```
Comment (philip-stoev) [inline misc/python/materialize/checks/scenarios_migration.py:24]: I think the Minipulate No2 should come after the migration. This ensures that DDLs and DMLs are run post-migration as well.

File: misc/python/materialize/mzcompose/services/schema_registry.py [hunk 3]
```diff
@@ -36,37 +37,37 @@ def __init__(
...
+                    "CMD",
+                    "curl",
+                    # We provide credentials in case the schema registry is
+                    # configured to require HTTP authentication, as there's
+                    # no health check endpoint that's excluded from
+                    # authentication requirements. The credentials are
+                    # safely ignored if the schema registry is not
+                    # configured to require them.
+                    "-fu",
+                    "materialize:sekurity",
```
Comment (philip-stoev) [inline misc/python/materialize/mzcompose/services/schema_registry.py:60]: This `sekurity` string may now be present at several places in the codebase. Consider replacing with a named constant.

Comment (philip-stoev) [reply misc/python/materialize/mzcompose/services/schema_registry.py:60]: Ops you just moved that code, nevermind.

**Review by philip-stoev — APPROVED** [review]
This is great -- Thank you!

I did not even realize that it is possible to string all the parts like that and achieve such a result.

File: misc/python/materialize/mzcompose/__init__.py [hunk 4]
```diff
@@ -29,7 +29,7 @@
 say = ui.speaker("C> ")
 
 
-DEFAULT_CONFLUENT_PLATFORM_VERSION = "7.5.0"
+DEFAULT_CONFLUENT_PLATFORM_VERSION = "7.5.2"
```
Comment (nrainer-materialize) [inline misc/python/materialize/mzcompose/__init__.py:32]: No need to change it this time, but I don't like it that much to mix different changes in a PR (unless this is required for this PR).

**Review by nrainer-materialize — APPROVED** [review]
LGTM besides the points already brought up by Philip

File: ci/plugins/mzcompose/hooks/command (hunk 1 above)
Comment (def-) [reply ci/plugins/mzcompose/hooks/command:83]: Oops, that was just me experimenting to see if qemu-static was working with docker, which was a requirement for this whole test.

File: misc/python/materialize/checks/scenarios_migration.py (hunk 2 above)
Comment (def-) [reply misc/python/materialize/checks/scenarios_migration.py:24]: I'll try! My main worry was that qemu-static is super slow, so I might have to change some deadlines.

File: misc/python/materialize/mzcompose/__init__.py (hunk 4 above)
Comment (def-) [reply misc/python/materialize/mzcompose/__init__.py:32]: I should have given an explanation. I had to check if these versions exist on Dockerhub for arm64, and while doing that thought it makes sense to use the latest version anyway. But yes, strictly speaking doesn't belong into this PR.

**Pull request merged**